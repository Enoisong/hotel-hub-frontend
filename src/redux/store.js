import { configureStore } from '@reduxjs/toolkit';
import reservationsReducer from './features/reservations/reservationsSlice';
import roomReducer from './features/rooms/roomsSlice';

const store = configureStore({
  reducer: {
    reservations: reservationsReducer,
    room: roomReducer,
  },
});

export default store;
