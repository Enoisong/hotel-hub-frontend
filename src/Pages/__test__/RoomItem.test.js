import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import Table from '../../components/Table/Table'; // Assuming the correct path

// Mock the localStorage
const localStorageMock = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  clear: jest.fn(),
};
global.localStorage = localStorageMock;

describe('RoomItem Component', () => {
  it('renders RoomItem component', () => {
    // Sample data for testing
    const tablehead = ['Name', 'Age'];
    const tablerows = [
      { name: 'John Doe', age: 25 },
      { name: 'Jane Doe', age: 30 },
    ];

    render(
      <Provider store={store}>
        <Router>
          <Table tablehead={tablehead} tablerows={tablerows} />
        </Router>
      </Provider>,
    );
  });

  // Add a snapshot test
  it('matches snapshot', () => {
    // Sample data for testing
    const tablehead = ['Name', 'Age'];
    const tablerows = [
      { name: 'John Doe', age: 25 },
      { name: 'Jane Doe', age: 30 },
    ];

    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <Table tablehead={tablehead} tablerows={tablerows} />
        </Router>
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
