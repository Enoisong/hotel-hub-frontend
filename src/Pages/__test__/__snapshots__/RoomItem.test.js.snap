// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomItem Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative flex flex-col bg-clip-border bg-white text-gray-700 shadow-md h-full w-full overflow-scroll rounded-none"
  >
    <table
      class="w-full min-w-max table-auto text-left"
    >
      <thead>
        <tr>
          <th
            class="border-b border-blue-gray-100 bg-blue-gray-50 p-4"
          >
            <p
              class="block antialiased font-sans text-sm text-blue-gray-900 font-normal leading-none opacity-70"
            >
              Name
            </p>
          </th>
          <th
            class="border-b border-blue-gray-100 bg-blue-gray-50 p-4"
          >
            <p
              class="block antialiased font-sans text-sm text-blue-gray-900 font-normal leading-none opacity-70"
            >
              Age
            </p>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="even:bg-blue-gray-50/50"
        >
          <td
            class="p-4"
          >
            <p
              class="block antialiased font-sans text-sm leading-normal text-blue-gray-900 font-normal"
            >
              John Doe
            </p>
          </td>
        </tr>
        <tr
          class="even:bg-blue-gray-50/50"
        >
          <td
            class="p-4"
          >
            <p
              class="block antialiased font-sans text-sm leading-normal text-blue-gray-900 font-normal"
            >
              Jane Doe
            </p>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
